doc/user/content/transform-data/strategies-considerations.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Strategies and considerations"
+description: ""
+menu:
+  main:
+    parent: "transform-data"
+    weight: 95
+    name: "Strategies and considerations"
+---
+
+Materialize provides various [indexing strategies to optimize query
+performance](/transform-data/optimization/). However, because indexes in
+Materialize maintain the complete query results in memory (not just the index
+keys and the pointers to data rows), ... [will be updated].
+
+The following lists some strategies and considerations that may help you better
+utilize the benefits provided by Materialize while managing
+[costs](/administration/billing/).
+
+## Strategies to reduce dataset/memory usage
```

Comment:
I thought I'd  surface late materialization as well but in the intro, I had initially linked to the optimization page where we mentioned late materialization. And even though I ended up removing that intro, left it out for now to keep the review of the list simpler.